pkg/operation/botanist/controlplane/konnectivity/konnectivity_server.go

```diff
@@ -0,0 +1,560 @@
+	ClientCA component.Secret
+}
+
+// ServerOptions are the options for the konnectivity-server.
+type ServerOptions struct {
+	// Client to create resources with.
+	Client client.Client
+	// Namespace in the seed cluster.
+	Namespace string
+	// Image of the konnectivity-server.
+	Image string
+	// Replica count of the konnectivity-server.
+	Replicas int32
+	// Hosts used for SNI.
+	Hosts []string
+	// IstioIngressLabels are the istio-ingressgateway's labels.
+	IstioIngressLabels map[string]string
+	// Healthy probes that konnectivity-server is healthy.
+	// Defaults to managedresources.WaitUntilManagedResourceHealthy.
+	Healthy Prober
+	// Removed probes that konnectivity-server is removed.
+	// Defaults to managedresources.WaitUntilManagedResourceDeleted.
+	Removed Prober
+}
+
+// NewServer creates a new instance of KonnectivityServer for the konnectivity-server.
+func NewServer(so *ServerOptions) (KonnectivityServer, error) {
+	if so == nil {
+		return nil, errors.New("serveroptions are required")
+	}
+
+	if so.Client == nil {
+		return nil, errors.New("client cannot be nil")
+	}
+
+	if len(so.Namespace) == 0 {
+		return nil, errors.New("namespace cannot be empty")
+	}
+
+	if so.Healthy == nil {
+		so.Healthy = managedresources.WaitUntilManagedResourceHealthy
+	}
+
+	if so.Removed == nil {
+		so.Removed = managedresources.WaitUntilManagedResourceDeleted
+	}
+
+	return &konnectivityServer{ServerOptions: so}, nil
+}
+
+// OpDestroy destroys components created by connectivity server instead of creating them.
+func OpDestroy(ks KonnectivityServer) KonnectivityServer {
+	return &opDestroy{component.OpDestroy(ks)}
+}
+
+type konnectivityServer struct {
+	*ServerOptions
+	secrets ServerSecrets
+}
+
+func (k *konnectivityServer) Deploy(ctx context.Context) error {
+	if k.secrets.Kubeconfig.Name == "" || k.secrets.Kubeconfig.Checksum == "" {
+		return errors.New("missing kubeconfig secret information")
+	}
+	if k.secrets.Server.Name == "" || k.secrets.Server.Checksum == "" {
+		return errors.New("missing server secret information")
+	}
+	if k.secrets.ClientCA.Name == "" || k.secrets.ClientCA.Checksum == "" {
+		return errors.New("missing client-ca secret information")
+	}
+	if len(k.Hosts) == 0 {
+		return errors.New("hosts cannot be empty")
+	}
+	if len(k.Image) == 0 {
+		return errors.New("image cannot be empty")
+	}
+	if len(k.IstioIngressLabels) == 0 {
+		return errors.New("istioingreslabels cannot be empty")
+	}
+
+	const (
+		name                          = ServerName
+		adminPort               int32 = 8133
+		healthPort              int32 = 8134
+		serverAgentPortUnsigned       = uint32(ServerAgentPort)
+		serverMountPath               = "/certs/konnectivity-server"
+		clientCAMountPath             = "/certs/client-ca"
+		kubeconfigMountPath           = "/auth"
+	)
+
+	var (
+		minAvailable = intstr.FromInt(1)
+		tcpProto     = corev1.ProtocolTCP
+		serviceHost  = fmt.Sprintf("%s.%s.svc.%s", name, k.Namespace, gardenercorev1.DefaultDomain)
```

Comment:
```suggestion
		konnectivityServerServiceHost  = fmt.Sprintf("%s.%s.svc.%s", name, k.Namespace, gardenercorev1.DefaultDomain)
```

Just makes it a bit easier to understand where it is routing to it in the VirtualService definition below.